#! /bin/sh -e
## 05_no_Xdefaults.dpatch by Eduard Bloch <blade@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Merge in-the-server resources after the distribution's ones

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -u -r1.40 xdefaults.C
--- a/src/xdefaults.C	22 Aug 2004 01:45:12 -0000	1.40
+++ b/src/xdefaults.C	28 Aug 2004 09:01:00 -0000
@@ -874,7 +874,7 @@
             }
         }
     }
-#   endif
+#   endif /* HAVE_EXTRA_XRESOURCE_FILES */
 
     /* Add in XENVIRONMENT file */
 
@@ -899,9 +899,16 @@
         if ((rdb1 = XrmGetFileDatabase (fname)) != NULL)
           XrmMergeDatabases (rdb1, &database);
       }
+    
+    // temporary workaround to restore user's custom xrdb settings
+    if (displayResource != NULL) {
+       rdb1 = XrmGetStringDatabase (displayResource);
+       if (rdb1 != NULL)
+          XrmMergeDatabases (rdb1, &database);
+    }
 
     XrmSetDatabase (display, database);
-#  endif
+#  endif /* XrmEnumOneLevel */
 
     /*
      * Query resources for options that affect us

